src/main/scala/mesosphere/marathon/core/instance/update/InstanceUpdater.scala

```diff
@@ -10,10 +10,37 @@ import mesosphere.marathon.core.task.update.TaskUpdateEffect
 import mesosphere.marathon.state.{Timestamp, UnreachableEnabled}
 import org.apache.mesos.{Protos => MesosProtos}
 
+trait InstanceExpungeStrategy {
```

Comment:
I guess this is an implementation detail then. However, I'd prefer a rescheduling strategy and have the definition for `schouledBeExpunged` be inferred from that. There is no need to change the goal. As long as we agree not to attach an expunge strategy to the run spec I'm fine with this change.

Reply:
Yeah, we should not have the expungeStrategy on the runSpec, a rescheduling strategy makes a lot more sense there. 
And I think having the custom ExpungeStrategies here, implementing different ReschedulingStrategies per runSpec gets a lot easier.